src/mail/mail.service.ts

```diff
@@ -60,53 +59,17 @@ export class MailService {
 				else{
 					err = null
 					attempts += 1
+                    delay(this.errorDelay)
 				}
 			}
 		}
 		// getting here means sending the mail is not possible => throw an error
-		throw new InternalServerErrorException("Unable to connect to mailserver")
+		throw new ServiceUnavailableException("Unable to connect to mailserver")
 	}
 
-	/**
-	 * Method called by the test endpoint
-	 * @param recipient - receiver for the email to be send
-	 * @param data - data to be displayed and rendered by ejs
-	 */
-	async mailTest(recipient: string, data: string): Promise<void> {
-		const tmpl = await readFile(__dirname + '/templates/test.ejs', 'utf-8')	
-		const message = render(tmpl, {
-			data: data
-		});
-		await this.sendMail(recipient, "test", message)
-	}
-
-	/**
-	 * Generate the verification Mail
-	 * @param name - username 
-	 * @param mail - user mail
-	 * @param code - verification code
-	 */
-	async generateVerifyMail(name: string, mail: string, code: string): Promise<void>{
-		const template = await readFile(join(__dirname, 'templates', 'MailVerify.ejs'), 'utf-8')
-		const message = render(template, {
-			verifyLink: `${process.env.HOST}/user/verify/${code}`,
-			username: name,
-		});
-		await this.sendMail(mail, "Please verify your email address", message)
-	}
-
-	/**
-	 * Sends a mail containing a password reset link/code
-	 * @param name - username
-	 * @param mail - user email
-	 * @param resetCode - the code required to reset/change the pw
-	 */
-	async sendPasswordReset(name: string, mail: string, resetCode: string): Promise<void>{
-		const tmpl = await readFile(__dirname + '/templates/PasswordReset.ejs', 'utf-8')	
-		const message = render(tmpl, {
-			resetLink: `${process.env.HOST}/user/reset-form/${resetCode}`,
-			username: name,
-		});
-		await this.sendMail(mail, "Reset your pw", message)
-	}
-}
\ No newline at end of file
+    async sendMail(recipient: string, template: string, mailData: (MailVerifyDto | MailPwResetDto | MailTestDto), subject: string ): Promise<void>{
```

Comment:
This is not modular. (We have to change sendMail to get a new mailData interface + we have to use as in the sendMail command)

We could use generics here: 
async sendMail<T>(.... mailData: T ....)

sendMail<MailVerifyDto>("support@gmail.com"....)